go/tools/builders/protoc.go

```diff
@@ -95,28 +104,46 @@ func run(args []string) error {
 		}
 	}
 	// Walk the generated files
-	filepath.Walk(*outPath, func(path string, f os.FileInfo, err error) error {
-		if !strings.HasSuffix(path, ".go") {
+	filepath.Walk(tmpDir, func(path string, f os.FileInfo, err error) error {
+		relPath, err := filepath.Rel(tmpDir, path)
+		if err != nil {
+			return err
+		}
+		if relPath == "." {
 			return nil
 		}
-		info := files[path]
-		if info != nil {
-			info.created = true
+
+		if f.IsDir() {
+			if err := os.Mkdir(abs(filepath.Join(*outPath, relPath)), f.Mode()); !os.IsExist(err) {
```

Comment:
Move the `abs` call to `*outPath` next to `tmpDir` above and use the result here. The comment can explain why both are needed.

Reply:
Done.